pkg/gui.py

```
@@ -190,152 +213,132 @@ def check_status(self, thread):
+            self.popup_widget.after(self.alive_time - 100, label.destroy)
+
             self.since_last_popup = time.now()
 
-            print("--Displayed popup--")
+            logger.debug("--Displayed popup--")
+        
+        elif message_item and "ERROR" in message_item[0]:
+            self.display_body_message(message_item[0])
 
-    def add_to_history(self, item):
+    def display_body_message(self, message: str, display_time: int = None) -> None:
```

Ruff (0.8.0)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)